docs/roce-perf-k8s.md

```diff
@@ -0,0 +1,374 @@
+
+   100GbE RoCE is used here to provide the interconect between two GPU worker nodes. RoCE provides the Remote Direct Memory Access (RDMA) across the Ethernet and IP network from one host to another with reduced latency and lower CPU overhead. Traditionally Ethernet and IP network are considered as a "lossy" network since they're not designed to provide an end-to-end loseless transmission, the packet drops occured during the transmission are supposed to be taken care of by upper layer protocoles. Roce remedies this by utilizing PFC and ECN to control the flow and managment the congestion and traffic transmission. We have taken following considerations into our implementation:
+
+   * Enable and configure RoCE properly wherever it's applicable:
+   	* Use RoCE NICs in servers with appropriate drivers installed. 
+   	* Use RoCE capable high performance non-blocking Ethernet switch supporting PFC and ECN based traffic management functions.
+   	* Software libraryies that can take advantage RoCE.
+   * LACP or any link bundle with multiple NICs are not recommendated since it doesn't work well with RoCE and RDMA in general. It's recommended to have each NIC on a seperate subnet.  
+   * Ideally each NIC should be on a separate "rail" to form a multi-rail topology. At the minimal, NICs on the same PCIe switch and NUMA node should spread out to different physical switches. please refer to server hardware block diagram on how the NICs and GPUs are connected internally.
+   * SRIOV with RoCE is the key technology to enable Kubernetes pod to achieve near line rate performance and low latency. Single Root I/O Virtualization (SRIOV) virtualize a single physical RoCE NIC into multiple virtual RoCE intrefaces (it's called VF in SRIOV's terminology) and directly attaches it to Kubernetes pod without going through the Linux kernel, so higher perforamnce and lower latency can be achieved because the entire data path is now bypassing the Linux kernel. 
+
+## Key Hardware & Software Requirements
+
+   * 1 x DGX-1 server used as both Kubernetes master node and DeepOps management node
+   * 2 x DGX-1 servers used as Kubernetes GPU worker nodes:
+   	* 8 x Tesla V100 GPU with total of 256GB (32GB x 8) HBM2 GPU memory
+   	* 2 x Mellanox ConnectX-5 VPI 100G NICs (configured as RoCE)
+   * 1 x Mellanox Spectrum SN2700 non-blocking 3.2T RoCE Ethernet switch
+   * Software components:
+   	* [NVIDIA DGX OS 4.4](https://docs.nvidia.com/dgx/dgx-os-server-release-notes/index.html#dgx-os-server-sw-versions)
+   	* Kubernetes v.1.15 and v.1.16 
+   	* latest MPI operator for Kubernetes 
+   	* latest Multus CNI to support multiple interfaces
+   	* latest SRIOV CNI and device plugin
+   	* latest Nvidia container toolkit
+   	* Nvidia NCCL 2.5.6 with CUDA 10.1
+   	* OpenMPI 3.1.5 and 4.0.0
+   	* TensorFlow 2.1.0
+   	* Mellanox Onyx v3.8 for SN2700
+   	* MOFED 4.7-3.2.9.0 and 4.6-1.0.1.1 for Mellanox ConnectX-5
+   * Internet access for software package upgrade
```

Comment:
This list isn't rendering correctly in Github, may need additional nesting of the sub-bullets?